addons/account/models/account_invoice.py

```diff
@@ -216,7 +216,7 @@ def _get_payments_vals(self):
                 amount_to_show = amount_currency
             else:
                 currency = payment.company_id.currency_id
-                amount_to_show = currency._convert(amount, self.currency_id, payment.company_id, self.date or fields.Date.today())
+                amount_to_show = currency._convert(amount, self.currency_id, payment.company_id, payment.date or fields.Date.today())
```

Comment:
isn't it payment.payment_date ?
https://github.com/odoo/odoo/blob/12.0/addons/account/models/account_payment.py#L778

Otherwise, this is consistent with what is done at https://github.com/odoo/odoo/blob/12.0/addons/account/models/account_invoice.py#L169